### PR TITLE
Changes to update all-in-one image

### PR DESCRIPTION
**Pull request by Rajalakshmi-Girish**
Fixes https://github.com/ppc64le-cloud/test-infra/issues/394

Successful run using updated image `quay.io/powercloud/all-in-one:0.6` - https://prow.ppc64le-cloud.cis.ibm.net/view/gs/ppc64le-kubernetes/logs/test-periodic-kubernetes-containerd-conformance-test-ppc64le/1685120647295930368

The issue above asks to update below things:
- [x] base debian image to latest stable image Bookworm -**DONE**
- [x] latest stable golang - **DONE** to `1.20.6`
- [x] latest terraform version - **DONE** to `v1.5.4`
- [x] latest providers like ibm, null etc.- **DONE** to `1.55.0` and `3.2.1` respectively
- [x] IBM Cloud cli and plugins - _cli **DONE** to `2.18.0` and as plugins are installed via `ibmcloud plugin install power-iaas` and ` ibmcloud plugin install cloud-object-storage`, they dont need update._
- [x] kubectl - **DONE** to `1.27.4`
- [ ] google sdk - _as using `wget -q https://dl.google.com/dl/cloudsdk/channels/rapid/google-cloud-sdk.tar.gz` URL, needn't change._

### CODE REVIEW

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: images/all-in-one/Dockerfile [hunk 1]
```diff
@@ -29,28 +29,28 @@ RUN git clone -b v${TERRAFORM_PROVIDER_NULL_VERSION} --single-branch https://git
...
     TERM=xterm
 ENV PATH /usr/local/go/bin:$PATH
 
-ARG KUBECTL_VERSION=v1.24.4
+ARG KUBECTL_VERSION=v1.27.4
 
 RUN apt-get update && apt-get install -y --no-install-recommends \
     apt-transport-https \
-    ca-certificates \
+    ca-certificates-java \
```
[the remark this answers is not in the document]
Comment (Rajalakshmi-Girish) [reply images/all-in-one/Dockerfile:53]: While building the image, there were failures with `jdk` install. It was mentioning the errors related to missing `ca-certificates-java`.
This change avoided the error and made the image build successful.
Hmm, I missed copying the error that I saw.

[the remark this answers is not in the document]
Comment (mkumatag) [reply images/all-in-one/Dockerfile:53]: can you please those errors, will be helpful to understand why we need this dependency?

[the remark this answers is not in the document]
Comment (Rajalakshmi-Girish) [reply images/all-in-one/Dockerfile:53]:
```
#0 222.0 Exception in thread "main" java.lang.InternalError: Error loading java.security file
#0 222.0        at java.base/java.security.Security.initialize(Security.java:106)
#0 222.0        at java.base/java.security.Security$1.run(Security.java:84)
#0 222.0        at java.base/java.security.Security$1.run(Security.java:82)
#0 222.0        at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
#0 222.0        at java.base/java.security.Security.<clinit>(Security.java:82)
#0 222.0        at java.base/sun.security.jca.ProviderList.<init>(ProviderList.java:178)
#0 222.0        at java.base/sun.security.jca.ProviderList$2.run(ProviderList.java:96)
#0 222.0        at java.base/sun.security.jca.ProviderList$2.run(ProviderList.java:94)
#0 222.0        at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
#0 222.0        at java.base/sun.security.jca.ProviderList.fromSecurityProperties(ProviderList.java:93)
#0 222.0        at java.base/sun.security.jca.Providers.<clinit>(Providers.java:55)
#0 222.0        at java.base/sun.security.jca.GetInstance.getInstance(GetInstance.java:156)
#0 222.0        at java.base/java.security.cert.CertificateFactory.getInstance(CertificateFactory.java:193)
#0 222.0        at org.debian.security.KeyStoreHandler.<init>(KeyStoreHandler.java:50)
#0 222.0        at org.debian.security.UpdateCertificates.<init>(UpdateCertificates.java:65)
#0 222.0        at org.debian.security.UpdateCertificates.main(UpdateCertificates.java:51)
#0 222.1 dpkg: error processing package ca-certificates-java (--configure):
#0 222.1  installed ca-certificates-java package post-installation script subprocess returned error exit status 1
............
..........
.........
#0 228.9 Errors were encountered while processing:
#0 228.9  ca-certificates-java
#0 229.3 E: Sub-process /usr/bin/dpkg returned an error code (1)
------
Dockerfile:60
--------------------
  59 |
  60 | >>> RUN apt-get update && apt-get install -y --no-install-recommends \
  61 | >>>     ansible \
  62 | >>>     build-essential \
  63 | >>>     ca-certificates \
  64 | >>>     curl \
  65 | >>>     default-jdk \
  66 | >>>     file \
  67 | >>>     git \
  68 | >>>     gcc \
  69 | >>>     jq \
  70 | >>>     make \
  71 | >>>     mercurial \
  72 | >>>     openssh-client \
  73 | >>>     pkg-config \
  74 | >>>     procps \
  75 | >>>     rsync \
  76 | >>>     unzip \
  77 | >>>     wget \
  78 | >>>     xz-utils \
  79 | >>>     zip \
  80 | >>>     zlib1g-dev \
  81 | >>>     && rm -rf /var/lib/apt/lists/*
  82 |
--------------------
ERROR: failed to solve: process "/bin/sh -c apt-get update && apt-get install -y --no-install-recommends     ansible     build-essential     ca-certificates     curl     default-jdk     file     git     gcc     jq     make     mercurial     openssh-client     pkg-config     procps     rsync     unzip     wget     xz-utils     zip     zlib1g-dev     && rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 100
[root@raji-workspace2 all-in-one]#
```
Please find the complete log from attached file 
[all-in-one_image_build_trace.txt](https://github.com/ppc64le-cloud/test-infra/files/12225175/all-in-one_image_build_trace.txt)

[the remark this answers is not in the document]
Comment (mkumatag) [reply images/all-in-one/Dockerfile:53]: may be it is happening for the package `default-jdk`, we can remove that package(I don't know who is using have with this container) and see how it goes.

[the remark this answers is not in the document]
Comment (Rajalakshmi-Girish) [reply images/all-in-one/Dockerfile:53]: I suppose this package was used when bazel was there in `all-in-one` image.
Will build, test and get back.

[the remark this answers is not in the document]
Comment (Rajalakshmi-Girish) [reply images/all-in-one/Dockerfile]: Yes, the removal of `default-jdk` made image building successful.
https://prow.ppc64le-cloud.cis.ibm.net/view/gs/ppc64le-kubernetes/logs/test-periodic-kubernetes-containerd-conformance-test-ppc64le/1686330566535811072 was run successfully with `all-in-one:0.6`
Please have a look at the change.

[the remark this answers is not in the document]
Comment (Rajalakshmi-Girish) [reply images/all-in-one/Dockerfile]: @mkumatag this comment is resolved.

**Review by mkumatag — APPROVED** [review]
/lgtm

Hope you have already built the image and pushed for both ppc and amd64 arch

**Pull request merged**